osu.Game/Screens/Play/Player.cs

```diff
@@ -627,6 +652,11 @@ public override void OnEntering(IScreen last)
 
             foreach (var mod in Mods.Value.OfType<IApplicableToHUD>())
                 mod.ApplyToHUD(HUDOverlay);
+
+            DrawableRuleset.IsPaused.BindValueChanged(_ => updateOverlayActivationMode());
```

Comment:
i'm pretty sure the state is not guaranteed to stop mutating when exiting player. probably needs a call to `updateOverlayActivationMode()` in exiting.

Reply:
I'm not sure at all with that one. Isn't the updating of the overlay mode with the change of screens sufficient?

Reply:
i.e. enabling the "disable overlays during gameplay" setting while in results screen will affect the global value, because `Player` still exists, just suspended.

Reply:
ah it's again that case. Guess I'll wait for your proposed changes to get merged and rebase, as it'll fix that case (aswell as the binding on entering)